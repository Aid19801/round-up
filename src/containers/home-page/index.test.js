import React from 'react';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import Homepage from './index';
import Title from '../../components/title';
import store from '../../store';

configure({ adapter: new Adapter() })

describe('Homepage', () => {
  
  it('should render the HomePage container without exploding...', () => {
      const wrapper = shallow(<Homepage />);
      expect(wrapper).toBeTruthy();
      expect(wrapper.contains(<Title />))
  });
  
  it('matches the snapshot', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
            <Homepage />
        </Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  })
});