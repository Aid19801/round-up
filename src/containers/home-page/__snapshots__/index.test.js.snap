// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Homepage matches the snapshot 1`] = `
<div
  className="home-page-container"
>
  <div
    className="title"
  >
    <h4>
      Round Up
    </h4>
  </div>
  <div
    className="container"
  >
    <div
      className="row"
    >
      <div
        className="col-sm-6 transactions-container"
      >
        <h4>
          Add To Savings: £
          0
        </h4>
        <button
          onClick={[Function]}
        >
          transfer =&gt; 
        </button>
        <div
          className="table-container"
        >
          <table
            className="table"
          >
            <thead>
              <tr>
                <th
                  scope="col"
                >
                  £/GBP
                </th>
                <th
                  scope="col"
                >
                  Balance
                </th>
                <th
                  scope="col"
                >
                  Source
                </th>
                <th
                  scope="col"
                >
                  IN / OUT
                </th>
                <th
                  scope="col"
                >
                  Date
                </th>
              </tr>
            </thead>
            <tbody />
          </table>
        </div>
      </div>
       
      <div
        className="col-sm-6 savings-container"
      >
        <h4>
          Savings Goal: £
          0
        </h4>
        <h4>
          current savings: £
          0
        </h4>
        <input
          className="goal-input"
          name="savingsGoal"
          onChange={[Function]}
          placeholder="enter your savings goal here..."
          type="text"
        />
        <button
          onClick={[Function]}
        >
          submit
        </button>
      </div>
    </div>
  </div>
</div>
`;
